mgmtfn/k8splugin/cniserver.go

```diff
@@ -0,0 +1,162 @@
+/***
+Copyright 2016 Cisco Systems Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package k8splugin
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/contiv/netplugin/mgmtfn/k8splugin/cniapi"
+	"github.com/contiv/netplugin/netplugin/plugin"
+	"github.com/gorilla/mux"
+)
+
+// contivKubeCfgFile holds credentials to access k8s api server
+const (
+	contivKubeCfgFile = "/opt/contiv/config/contiv.json"
+)
+
+// ContivConfig holds information passed via config file during cluster set up
+type ContivConfig struct {
+	K8sAPIServer string `json:"K8S_API_SERVER,omitempty"`
```

Comment:
Right now it's hard coded to v1 like this:
const (
        nsURL = "/api/v1/namespaces/"
)
We can make it configurable, if necessary.